### PR TITLE
tech detection: SAST Fixes (SonarLint)

### PR DESCRIPTION
**Pull request by kingthorin**
## Overview
- AppPattern > Remove unnecessary right side type declaration.
- CHANGELOG > Add maint note.
- ExtensionWappalyzer > Remove unused/unnecessary variables.
- PopupMenuEvidence > Use .equals() for String comparison.
- TechOptionsPanel > Use imports vs. fully qualified objects.
- TechPanel > Adjust to more appropriate use of constants, use imports vs. fully qualified objects. Remove use of statically assigned/sized fonts.
- TechPassiveScanner > Remove unnecessary right side type declaration.
- TechJsonParser > Remove unnecessary ThreadGroup declaration and assignment.

## Related Issues
N/A

## Checklist
- [na] Update help
- [x] Update changelog
- [x] Run `./gradlew spotlessApply` for code formatting
- [na] Write tests
- [na] Check code coverage
- [x] Sign-off commits
- [x] Squash commits
- [x] Use a descriptive title

### CODE REVIEW

**Review by psiinon — APPROVED** [review]

File: addOns/wappalyzer/src/main/java/org/zaproxy/zap/extension/wappalyzer/TechPanel.java [hunk 1]
```diff
@@ -157,7 +160,7 @@ private javax.swing.JToolBar getPanelToolbar() {
             panelToolbar.setFloatable(false);
             panelToolbar.setRollover(true);
             panelToolbar.setPreferredSize(new java.awt.Dimension(800, 30));
-            panelToolbar.setFont(new java.awt.Font("Dialog", java.awt.Font.PLAIN, 12));
+            panelToolbar.setFont(new Font(Font.DIALOG, Font.PLAIN, 12));
```
Comment (thc202) [inline addOns/wappalyzer/src/main/java/org/zaproxy/zap/extension/wappalyzer/TechPanel.java:163]: This should actually be using `FontUtils`, or drop the hardcoded sizes, which are most likely not needed as the defaults are good elsewhere. (same for other places)

Comment (kingthorin) [reply addOns/wappalyzer/src/main/java/org/zaproxy/zap/extension/wappalyzer/TechPanel.java:163]: Good point, it probably doesn't need to be set at all. I'll check.

**Comment by kingthorin** [comment]
Seems fine without the static font settings

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Pull request merged**